### PR TITLE
Bump version to 0.1.25

### PR DESCRIPTION
**Pull request by zanieb**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,42 @@
...
+### Breaking changes
+
+- Limit overrides and constraints to `requirements.txt` format ([#2632](https://github.com/astral-sh/uv/pull/2632))
+
+### Enhancements
+
+- Accept `setup.py` and `setup.cfg` files in compile ([#2634](https://github.com/astral-sh/uv/pull/2634))
+- Add `--no-binary` and `--only-binary` support to `requirements.txt` ([#2680](https://github.com/astral-sh/uv/pull/2680))
+- Allow prereleases, locals, and URLs in non-editable path requirements ([#2671](https://github.com/astral-sh/uv/pull/2671))
+- Use PEP 517 to extract non-static `pyproject.toml` metadata ([#2633](https://github.com/astral-sh/uv/pull/2633))
```
Comment (AlexWaygood) [inline CHANGELOG.md:14]: AKA?

```suggestion
- Use PEP 517 to extract dynamic `pyproject.toml` metadata ([#2633](https://github.com/astral-sh/uv/pull/2633))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,42 @@
...
+- Add `Editable project location` and `Required-by` to `pip show` ([#2589](https://github.com/astral-sh/uv/pull/2589))
+- Avoid `prepare_metadata_for_build_wheel` calls for Hatch packages with dynamic dependencies ([#2645](https://github.com/astral-sh/uv/pull/2645))
+- Fall back to PEP 517 hooks for non-compliant PEP 621 metadata ([#2662](https://github.com/astral-sh/uv/pull/2662))
+- Support `file://localhost/` schemes ([#2657](https://github.com/astral-sh/uv/pull/2657))
+- Use normal resolver in `pip sync` ([#2696](https://github.com/astral-sh/uv/pull/2696))
+
+### CLI
+
+- Disallow `pyproject.toml` from `pip uninstall` ([#2663](https://github.com/astral-sh/uv/pull/2663))
```
Comment (AlexWaygood) [inline CHANGELOG.md:23]:
```suggestion
- Disallow `pyproject.toml` from `pip uninstall -r` ([#2663](https://github.com/astral-sh/uv/pull/2663))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,42 @@
...
+- Unhide `--emit-index-url` and `--emit-find-links` ([#2691](https://github.com/astral-sh/uv/pull/2691))
+- Use dense formatting for requirement version specifiers in diagnostics ([#2601](https://github.com/astral-sh/uv/pull/2601))
+
+### Performance
+
+- Add an in-memory cache for Git references ([#2682](https://github.com/astral-sh/uv/pull/2682))
+- Do not force recompile `.pyc` files ([#2642](https://github.com/astral-sh/uv/pull/2642))
```
Comment (AlexWaygood) [inline CHANGELOG.md:30]:
```suggestion
- Do not force-recompile `.pyc` files ([#2642](https://github.com/astral-sh/uv/pull/2642))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,42 @@
...
+- Read package metadata from `pyproject.toml` when statically defined ([#2676](https://github.com/astral-sh/uv/pull/2676))
```
Comment (AlexWaygood) [inline CHANGELOG.md:31]:
```suggestion
- Read package metadata from `pyproject.toml` when it is statically defined ([#2676](https://github.com/astral-sh/uv/pull/2676))
```

**Review by AlexWaygood — APPROVED** [review]

**Review by charliermarsh — APPROVED** [review]
I was initially hoping to include https://github.com/astral-sh/uv/pull/2702 but probably best to put that off for the next release anyway.

**Pull request merged**